imageproxy.go

```diff
@@ -403,7 +403,7 @@ func (t *TransformingTransport) RoundTrip(req *http.Request) (*http.Response, er
 	imageRequest, _ := http.NewRequest("GET", u.String(), nil)
 	imageRequest.Header = req.Header
 	resp, err := t.CachingClient.Do(imageRequest)
-	if err != nil {
+	if err != nil || resp.StatusCode >= 400 {
 		return nil, err
```

Comment:
Looking at this again, we may want to split this into 2 separate if conditions. Leave the previous condition as it is, and move the new check to it's own if. From there we can return a crafted error response that includes at minimum the status code. With the way it is now, what may happen is that the error will be `nil` and the response will be `nil`, which can make it difficult to debug what is going wrong in these scenarios.